android/tangram/src/com/mapzen/tangram/MapController.java

```diff
@@ -51,6 +54,52 @@
         void onFeaturePick(Map<String, String> properties, float positionX, float positionY);
     }
 
+    public interface ViewCompleteListener {
+        /**
+         * Called on the render-thread at the end of whenever the view is fully loaded and
+         * no ease- or label-animation is in running.
+         */
+        void onViewComplete();
+    }
+
+    public interface FrameCaptureCallback {
+        void onCaptured(Bitmap bitmap);
+    }
+
+    /**
+     * Capture MapView as Bitmap.
+     * @param waitForCompleteView Delay the capture until the view is fully loaded and
+     *                            no ease- or label-animation is in running.
+     */
+    public void captureFrame(FrameCaptureCallback callback, boolean waitForCompleteView) {
+        frameCaptureCallback = callback;
+        frameCaptureAwaitCompleteView = waitForCompleteView;
+        requestRender();
+    }
+
+    public Bitmap capture() {
+        // FIXME Check if on render thread!
+        int w = mapView.getWidth();
+        int h = mapView.getHeight();
+
+        int b[] = new int[(int) (w * h)];
+        int bt[] = new int[(int) (w * h)];
+        IntBuffer buffer = IntBuffer.wrap(b);
+        buffer.position(0);
+        GLES20.glReadPixels(0, 0, w, h, GLES20.GL_RGBA, GLES20.GL_UNSIGNED_BYTE, buffer);
```

Comment:
What is the reason of not reading the raw pixels in a native call? It would be the same for other platforms (and could still use platform image types like `Bitmap` and `CImage`.